src/main/scala/is/hail/expr/AST.scala

```diff
@@ -662,8 +662,16 @@ case class Apply(posn: Position, fn: String, args: Array[AST]) extends AST(posn,
 
   def toIR(agg: Option[String] = None): Option[IR] = {
     fn match {
-      case "merge" | "select" | "drop" | "annotate" | "index" =>
+      case "merge" | "select" | "drop" | "index" =>
```

Comment:
@cseed why are these specially called out? They should always fail to be found in the function registry nor are they primitive operations.

Reply:
I think this has to do with incomplete typechecking of the arguments of these methods.